system/ee/ExpressionEngine/Service/Addon/Installer.php

```diff
@@ -63,6 +63,21 @@ public function install()
      */
     public function update($current = '')
     {
+        if ($current == '' or version_compare($current, $this->version, '==')) {
+            return false;
+        }
+        
+        foreach ($this->actions as $action) {
+            if (!isset($action['class'])) {
+                $action['class'] = $classname;
+            }
+            $model = ee('Model')->get('Action')->filter('class', $action['class'])->filter('method', $action['method'])->first();
```

Comment:
@intoeetive I'm not sure I understand why this is needed in the add-on installer service?

Reply:
This is addressing really separate issue, which has been discovered while adding actions for MFA. The Installer service did not work correct when adding actions after it has been installed, this is fixing its behaviour (so you can add actions to the array and it will be updated; not necessary to write your own update code)